machine.js

```diff
@@ -0,0 +1,337 @@
+                tstates_callbacks.forEach(entry => {
+                    if (entry.tstates <= t_state) {
+                        tstatesutils.addTstates.in_callback = true;
+                        entry.callback();
+                        if (entry.period == 0) {
+                            tstates_callbacks.delete(entry);
+                        } else {
+                            entry.tstates += entry.period;
+                        }
+                        tstatesutils.addTstates.in_callback = false;
+                    }
+                });
+            }
+        },
+
+        /**
+         * Get the earliest callback out of the list.
+         */
+        getEarliestCallback: function() {
+            var earliest = null;
+
+            tstates_callbacks.forEach((entry) => {
+                if (earliest == null || entry.tstates < earliest.tstates) {
+                    earliest = entry;
+                }
+            });
+
+            return earliest;
+        },
+
+        /**
+         * Register a callback that shall be called after the number of T-states
+         * of the CPU given.
+         * If the given number is less than 0, return an error.
+         */
+        registerTstateCallback: function(callback, call_tstates) {
+            if (call_tstates < 0) {
+                return null;
+            }
+
+            var obj = null;
+
+            /* If the CPU is halted, not registering this event in the list
+            * will make us completely miss it when function getEarliestCallback()
+            * is called. Because of that, the CPU will miss this interrupt/event.
+            * Keeping call_tstates as 0 should work, but let's be safe and make it
+            * happen in the upcoming T-state.  */
+            if (call_tstates == 0) {
+                call_tstates = 1;
+            }
+
+            obj = { tstates: t_state + call_tstates, callback, period: 0 };
+            tstates_callbacks.add(obj);
+
+            return obj;
+        },
+
+        /* Register a callback to be called every call_tstates T-states.
+        * The delay parameter will let us, defer the start of the first call,
+        * without altering the period. This is handy for period signal that changes
+        * values for a short period of time (pulses)
+        */
+        registerTstateInterval: function(callback, call_tstates, delay) {
+            if (call_tstates < 0) {
+                return null;
+            }
+
+            /* If the delay parameter is not defined, set it to 0 */
+            delay = delay || 0;
+
+            const obj = { tstates: t_state + delay + call_tstates, callback, period: call_tstates };
+            tstates_callbacks.add(obj);
+            return obj;
+        },
+
+        removeTstateCallback: function(callback) {
+            if (callback != null) {
+                tstates_callbacks.delete(callback);
+            }
+        },
+
+        interrupt: function(interrupt_vector) {
```

Comment:
This function is not related to T-States, it should be taken out

Reply:
Fixed.